zhmc_prometheus_exporter/zhmc_prometheus_exporter.py

```diff
@@ -387,32 +408,39 @@ def create_session(cred_dict):
     Takes a dictionary with the HMC IP, the user ID, and a password.
     Returns the session.
     """
+
     # These warnings do not concern us
     urllib3.disable_warnings()
+
     session = zhmcclient.Session(cred_dict["hmc"],
                                  cred_dict["userid"],
-                                 cred_dict["password"])
+                                 cred_dict["password"],
+                                 retry_timeout_config=RETRY_TIMEOUT_CONFIG)
     return session
 
 
-def get_hmc_version(session, hmccreds_filename):
+def get_hmc_version(session):
     """
     Return the HMC version as a string "v.r.m".
+
+    Raises: zhmccclient exceptions
     """
-    with zhmc_exceptions(session, hmccreds_filename):
```

Comment:
Q: Why does this contextmanager become unnecessary?

Reply:
The context managers for transforming the exceptions are now on the caller's side. That makes the functions more flexible, I think. So overall, it is still used.

Reply:
Thank you! LGTM.